pkg/apis/wksprovider/machine/os/os.go

```diff
@@ -367,6 +385,103 @@ func (o OS) CreateSeedNodeSetupPlan(params SeedNodeParams) (*plan.Plan, error) {
 	return createPlan(b)
 }
 
+// Sets the pod CIDR block in the weave-net manifest
+func setWeaveNetPodCIDRBlock(manifests [][]byte, podsCIDRBlock string, cni string) ([][]byte, error) {
+	// Weave-Net has a container named weave in its daemonset
+	containerName := "weave"
+	// The pod CIDR block is set via the IPALLOC_RANGE env var
+	podCIDRBlock := &v1.EnvVar{
+		Name:  "IPALLOC_RANGE",
+		Value: podsCIDRBlock,
+	}
+
+	manifestList := &v1.List{}
+	err := yaml.Unmarshal(manifests[0], manifestList)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal weave-net manifest")
+	}
+
+	// Find and parse the DaemonSet included in the manifest list into an object
+	daemonSet, err := findDaemonSet(manifestList)
+	if err != nil {
+		return nil, errors.New("failed to find daemonset in weave-net manifest")
+	}
+
+	daemonSet.Spec.Template.Spec.Containers, err = injectEnvVarToContainer(
+		daemonSet.Spec.Template.Spec.Containers, containerName, *podCIDRBlock)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to inject env var to weave container")
+	}
+
+	daemonSetWithCIDRBlocks, err := yaml.Marshal(daemonSet)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal daemonset")
+	}
+
+	// First remove the default daemonset from the manifest list
+	manifestList.Items = manifestList.Items[:len(manifestList.Items)-1]
```

Comment:
This is assuming the Daemonset to replace is the last one in the list. It would be better to return the index of the daemonset from findDaemonSet and ensure you are removing the correct one here.

Reply:
Good catch, thanks. Changed the function to return the idx where the manifest was found, and shuffled the list in the unit test to test this.